packages/subgraph/TimeLock/src/mapping.ts

```diff
@@ -3,48 +3,48 @@ import {
   TimeLock,
   CallExecuted,
   CallScheduled,
-  Cancelled,
-  MinDelayChange,
-  RoleAdminChanged,
-  RoleGranted,
-  RoleRevoked
+  Cancelled
 } from "../generated/TimeLock/TimeLock"
-import { ProposalItem } from "../generated/schema"
+import { Proposal, Operation } from "../generated/schema"
 
 export function handleCallExecuted(event: CallExecuted): void {
-  let entity = ProposalItem.load(event.params.id.toHex())
-    if (entity == null) {
-      return
-    }
-  entity.status = 2
+  let entity = Proposal.load(event.params.id.toHex())
+  if (entity == null) {
+    return
+  }
+  entity.status = 'Executed'
   entity.save()
 }
 
 export function handleCallScheduled(event: CallScheduled): void {
-    let entity = new ProposalItem(event.params.id.toHex())
-    if (entity == null) {
-      return
-    }
-    entity.index = event.params.index
-    entity.target = event.params.target
-    entity.value = event.params.value
-    entity.data = event.params.data
-    entity.predecessor = event.params.predecessor
-    entity.delay = event.params.delay
-    entity.timestamp = event.block.timestamp
-  // 0 --- pending
-  // 1 --- ready
-  // 2 --- done
-  // 3 --- cancel
-    entity.status = 0
-    entity.save()
+  let proposal = Proposal.load(event.params.id.toHex())
+  if (proposal == null) {
+    proposal = new Proposal(event.params.id.toHex())
+    proposal.predecessor = event.params.predecessor
+    proposal.delay = event.params.delay
+    proposal.timestamp = event.block.timestamp
+    proposal.status = 'Pending'
+    proposal.save()
+  }
+
+  let id = event.params.id.toHex()
+  let index = event.params.index.toHex()
+  let op_id = id + index
+  let operation = new Operation(op_id)
+  operation.index = event.params.index
+  operation.target = event.params.target
+  operation.value = event.params.value
+  operation.data = event.params.data
+  operation.proposal = proposal.id
+  operation.save()
 }
 
 export function handleCancelled(event: Cancelled): void {
-  let entity = ProposalItem.load(event.params.id.toHex())
-    if (entity == null) {
-      return
-    }
-  entity.status = 3
+  let entity = Proposal.load(event.params.id.toHex())
+  if (entity == null) {
+    return
+  }
+  entity.status = 'Cancelled'
+  entity.timestamp = BigInt.zero()
```

Comment:
is it better to set timestamp to block.timestamp?

Reply:
https://github.com/darwinia-network/darwinia-bridge-sol/blob/dc35bc7574d2c5f383506b2fa51539531046688c/contracts/dao/flat/solc-0.8.6/TimeLock.f.sol#L716
I think `set it to 0` is the more in line with the contract design.